nautobot/extras/models/groups.py

```diff
@@ -495,13 +495,36 @@ def generate_query_for_filter(self, filter_field, value):
         """
         query = models.Q()
 
+        field_name = filter_field.field_name
+
+        # Attempt to account for `ModelChoiceFilter` where `to_field_name` MAY be set.
+        to_field_name = getattr(filter_field.field, "to_field_name", None)
+        if to_field_name is not None:
+            field_name = f"{field_name}__{to_field_name}"
+
+        lookup = f"{field_name}__{filter_field.lookup_expr}"
+
         # Explicitly call generate_query_{filter_method} for a method filter.
         if filter_field.method is not None and hasattr(filter_field.parent, "generate_query_" + filter_field.method):
             filter_method = getattr(filter_field.parent, "generate_query_" + filter_field.method)
-            query |= models.Q(filter_method(value))
+            query |= filter_method(value)
+
+        # Explicitly call `filter_field.generate_query` for a reversible filter.
+        elif hasattr(filter_field, "generate_query"):
+            filter_method = getattr(filter_field, "generate_query")
+
+            # Is this a list of strings? Well let's resolve it to related model objects so we can
+            # pass it to `generate_query` to get a correct Q object back out.
+            if value and isinstance(value, list) and isinstance(value[0], str):
+                related_model = self.model._meta.get_field(filter_field.field_name).related_model
+                lookup_kwargs = {f"{to_field_name}__in": value}
```

Comment:
This assertion is fragile. A list of strings might not necessarily coincide with a filter that has `to_field_name` set. This will be shaken out in testing.

Reply:
I'm realizing that this also may have implications for `SearchFilter`, since it also implements `generate_query` but its signature is different. Good times.